### ISSUE TITLE
Add item filtering by data attributes

### ISSUE THREAD

**Issue by domialex** [domialex/Sidekick#8]

**Comment by nielsuiterwijk** [comment]
I've been looking through the code, and I have a proof of concept working now. It's a matter of finding all the correct api filters, such as `pseudo.pseudo_total_cold_resistance`. 

I don't have much time in the evenings, but I could push something to a branch and have other people work on it as well. In my day job i write C++ so I might also be writing horribly inefficient C# code..

**Comment by domialex** [comment]
Ok so since we have the static data `stats`. The `Item` object should, in my opinion, have a list of attributes.
There are 9 types:
```
Pseudo
Implicit
Fractured
Enchant
Crafted
Veiled
Monster
Delve
```
I didn't test every case, but it seems **Implicit** and **Crafted** are always added at the end of the line :
`50% increased Critical Strike Chance (implicit)`
`75% increased Mine Damage (crafted)`
This is not the case for anointments and enchants for example, which fortunately are treated as the same attribute https://i.imgur.com/u0rz90c.png https://i.imgur.com/ZOVPkeR.png.
It should be [relatively simple](https://dotnetfiddle.net/J3rx5d) to populate all those attributes in the item object, and when we call `GetListings` with this item, it should take into consideration the default attributes chosen in the user settings.
The attribute object should support min/max values and I know there might be some exceptions like `Increased Mana Reserved` which is the same attribute for `Reduced Mana Reserved` but with negative values.

**Comment by leMicin** [comment]
This is how it looks currently. Some features are missing, but seperate tasks were created for them (#287 #288 #289 )
![image](https://user-images.githubusercontent.com/5131398/77031308-94ff3780-6977-11ea-80cd-136c22eaecdf.png)

**Comment by leMicin** [comment]
Fixed by #243
